zerver/test_external.py

```diff
@@ -225,3 +226,16 @@ def test_change_user(self):
         result = self.client.post('/json/users/me/android_gcm_reg_id', {'token':token})
         self.assert_json_success(result)
 
+class SanitizeNameTests(TestCase):
+    def test_file_name(self):
+        fileName=sanitize_name('test.txt')
+        hiddenFile=sanitize_name('.hidden')
+        hiddenFileWithExtension=sanitize_name('.hidden.txt')
+        tarball=sanitize_name('tarball.tar.gz')
+        hiddenTarball=sanitize_name('.hiddentarball.tar.gz')
+        self.assertEquals(fileName, 'test.txt')
+        self.assertEquals(hiddenFile, '.hidden')
+        self.assertEquals(hiddenFileWithExtension, '.hidden.txt')
+        self.assertEquals(tarball, 'tarball.tar.gz')
+        self.assertEquals(hiddenTarball, '.hiddentarball.tar.gz')
+
```

Comment:
You can write this with half as many lines by just doing e.g.

`self.assertEquals(sanitize_name('test.txt'), 'test.txt')`

Also, you should add some test cases here where the code actually modifies the filenames in the various ways its expected to do that!  Otherwise, these tests don't confirm very much.

Finally, the code for your modified `sanitize_name` function doesn't actually achieve the goal of sanitizing the name -- all sorts of bad characters could be after the first "." in the filename.  So I think you want to still be calling `slugify` or something similar on the extension part...